// use this to compile (in remote directory directory):
// requires pip install grpcio-tools
// python -m grpc_tools.protoc -I protos --python_out=. --grpc_python_out=. protos/examples/blender_shadow/remote/blender_shadow.proto
syntax = "proto3";

// split out general structures using imports
package blender_shadow_worker;

message Numpy {
  string dtype = 1;
  repeated int32 shape = 2;
  bytes data = 3;
}

message Request {
  string id = 1;
  float target_x = 2;
  float target_y = 3;
}

message Response {
  string id = 1;
  Numpy render = 2;
}

service BlenderShadow {
  rpc GetVal(Request) returns (Response) {}
}
